src/web_app/js/background.js

```diff
@@ -9,14 +9,149 @@
 /* More information about these options at jshint.com/docs/options */
 
 // Variables defined in and used from chrome.
-/* globals chrome */
+/* globals chrome, sendAsyncUrlRequest, Constants */
 
 'use strict';
-chrome.app.runtime.onLaunched.addListener(function() {
-  chrome.app.window.create('appwindow.html', {
-    'width': 800,
-    'height': 600,
-    'left': 0,
-    'top': 0
+(function() {
+  chrome.app.runtime.onLaunched.addListener(function() {
+    chrome.app.window.create('appwindow.html', {
+      'width': 800,
+      'height': 600,
+      'left': 0,
+      'top': 0
+    });
   });
-});
+
+  // Send message from background to window.
+  var sendMessageToWindow = function(port, message) {
+    trace('B -> W: ' + JSON.stringify(message));
+    try {
+      port.postMessage(message);
+    }
+    catch (ex) {
+      trace('Error sending message: ' + ex);
+    }
+  };
+
+  var handleWebSocketRequest = function(port, message) {
+    if (message.wsAction === Constants.WS_CREATE_ACTION) {
+      trace('RWS: creating web socket: ' + message.wssUrl);
+      var ws = new WebSocket(message.wssUrl);
+      ws.onopen = function() {
+        trace('RWS: onopen');
+        sendMessageToWindow(port, {
+          action: Constants.WS_ACTION,
+          wsAction: Constants.EVENT_ACTION,
+          wsEvent: Constants.WS_EVENT_ONOPEN
+        });
+      };
+      ws.onerror = function() {
+        trace('RWS: onerror.');
+        sendMessageToWindow(port, {
```

Comment:
nit: make sendMessageToWindow take arguments |action|, |wsAction|, |wsEvent| and |data|?